packages/grid/x-data-grid-pro/src/components/DataGridProVirtualScroller.tsx

```diff
@@ -274,8 +325,88 @@ const DataGridProVirtualScroller = React.forwardRef<
 
   const detailPanels = getDetailPanels();
 
+  const topPinnedRows = getRows({ renderContext, rows: topPinnedRowsData });
+
+  const topPinnedRowsHeight = React.useMemo(
```

Comment:
I found an issue related to the height of the tob or bottom pinned rows render container. The issue is present when you change the density of the grid so you would need to also recompute this value when the density changes.
You can also test it the same thing will be present when you have dynamic or variable row height.

https://user-images.githubusercontent.com/5858539/178507706-c9a3e692-25d4-4945-b957-12067a5378f5.mov